### ISSUE TITLE
Shippable CI sunset

### ISSUE THREAD

**Issue by jforissier** [OP-TEE/optee_os#4367]
Shippable CI will no longer be available as of May 3rd 2021 as mentioned [here](https://blog.shippable.com/the-next-step-in-the-evolution-of-shippable-jfrog-pipelines). The purpose of this issue if to track the migration of `optee_os` to some other CI solution. It could be JFrogs Pipelines as mentioned in the blog post, or it could be something else.

**Comment by jforissier** [comment]
A side note: this could be a good opportunity to get rid of Travis CI too. Travis is now used only for coding style checks (`checkpatch` and `pycodestyle`), and booting a VM just for this is a bit overkill. Especially since our jobs tend to remain stuck in the "queued" state a bit too often for my taste :open_mouth:

**Comment by jbech-linaro** [comment]
Agree, Travis isn't what it used to be. Maybe time to polish IBART :) Or give Linaro CI a try again ...

**Comment by jenswi-linaro** [comment]
IBART is a good complement, but I think we need at least one other CI service.

**Comment by jbech-linaro** [comment]
> IBART is a good complement, but I think we need at least one other CI service.

Agree, a complement, is a good way to look at IBART. It gets the job done for some things, but it has some rough corners, limited capacity.

**Comment by Emantor** [comment]
We have been using [Github Actions](https://github.com/features/actions) since the travis shutdown, maybe it fits OP-TEE as well.

**Comment by jforissier** [comment]
@Emantor thanks for sharing, I will look at it. Do you know if GitHub Actions can run a Docker image?

**Comment by Emantor** [comment]
Yes, see the [docs](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action), we are using docker for [RAUC](https://github.com/rauc/rauc/blob/4da03a82cd162e3d155c9dec8c8dd94199dcef14/.github/workflows/tests.yml#L8).

**Comment by igoropaniuk** [comment]
U-Boot folks are using Azure for that. I was asked yesterday to run some tests there for one of my patch series, so I had to create account/setup pipeline, which barely took me around 5 minutes (btw, also uses similar approach with YAML config file as in Shippable, take a look https://gitlab.denx.de/u-boot/u-boot/-/blob/master/.azure-pipelines.yml)

Taking into account that for some pipelines average elapsed time is ~2hours, might be a good replacement for Shippable
https://dev.azure.com/bmeng/GitHub/_build/results?buildId=321&view=results

**Comment by jforissier** [comment]
So I looked at the documentation for both GitHub Actions and Microsoft Azure DevOps (Pipelines), and Azure looked easier (closer to the single `.shippable.yml` file we currently have). It was indeed super easy to setup and I have created https://github.com/OP-TEE/optee_os/pull/4421. Build time is comparable to Shippable but Open Source projects are entitled to 10 parallel builds for free which is quite cool.
